### PR TITLE
Potential enhancement to add any freetext as display name

### LINKED ISSUE #14: Add support for parsing email including name
I would expect during parsing we could extract the display name, similar to how we extract local_part and domain, such that the following would work:

```rust
use std::str::FromStr;
use email_address::EmailAddress;

fn main() {
    let email = "art@vandelay.com";
    let addr = EmailAddress::from_str(email).unwrap();
    println!("{:?}", addr);


    let email2 = "Art Vandelay <art@vandelay.com>";
    let addr2 = EmailAddress::from_str(email2).unwrap();
    println!("{:?}", addr2);
}
```

Current output:

```txt
EmailAddress("art@vandelay.com")
thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: InvalidCharacter', src/main.rs:11:48
```

### PR DESCRIPTION
**Pull request by ghandic**
closes #14

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -624,17 +678,38 @@ fn is_uri_reserved(c: char) -> bool {
...
+    let (local_part, domain) = split_at(email)?;
+    return Ok((local_part, domain, display))
+
+}
+
+fn split_display_email(text: &str) -> Result<(&str, &str), Error> {
+    match text.rsplit_once(" <") {
+        None => Ok(("", text)),
+        Some(left_right) => {
+            let email = &left_right.1[0..left_right.1.len() - 1];
```
Comment (johnstonskj) [inline src/lib.rs:721]: I assume the -1 is to drop the final `'>'` but this doesn't work if padded. I would make sure we call trim() on both parts before returning

**Review by johnstonskj — CHANGES_REQUESTED** [review]

File: src/lib.rs (hunk 1 above)
Comment (1oglop1) [reply src/lib.rs:721]: @ghandic Any progress on this?

Comment (johnstonskj) [reply src/lib.rs:721]: I thought it was bounced back to you to update the PR

**Review by johnstonskj — APPROVED** [review]

**Comment by johnstonskj** [comment]
Will put up a separate commit to merge this

**Pull request closed without merging**